delfin/drivers/hpe/hpe_msa/ssh_handler.py

```diff
@@ -0,0 +1,465 @@
+                                    get("serial-number")
+                    data_map = {
+                        'name': data.get('volume-name', ''),
+                        'storage_id': storage_id,
+                        'description': data.get('volume-name', ''),
+                        'status': status,
+                        'native_volume_id': str(data.get('durable-id', '')),
+                        'native_storage_pool_id': native_storage_pool_id,
+                        'wwn': str(data.get('wwn', '')),
+                        'type': data.get('volume-type', ''),
+                        'total_capacity': int(total_size),
+                        'free_capacit': int(total_avail),
+                        'used_capacity': int(total_size - total_avail),
+                        'blocks': int(data.get("blocks")),
+                        'compressed': True,
+                        'deduplicated': True
+                    }
+                    list_volumes.append(data_map)
+            return list_volumes
+        except Exception as e:
+            err_msg = "Failed to get storage volume: %s" % (six.text_type(e))
+            LOG.error(err_msg)
+            raise e
+
+    def list_storage_pools(self, storage_id):
+        try:
+            pool_infos = self.ssh_pool.do_exec('show pools')
+            pool_detail = self.handle_detail(pool_infos, 'pools')
+            volume_arr = self.list_storage_volume(storage_id)
+            pool_arr = []
+            if pool_detail:
+                for data in pool_detail:
+                    volume_size = 0
+                    blocks = 0
+                    for volume in volume_arr:
+                        if volume.get("native_storage_pool_id") == data.\
+                                get("serial-number"):
+                            volume_size += volume.get("total_capacity")
+                            blocks += volume.get("blocks")
+                    health = data.get('health', '')
+                    status = constants.StoragePoolStatus.OFFLINE
+                    if health == 'OK':
+                        status = constants.StoragePoolStatus.NORMAL
+                    total_size = self.parse_string(data.get('total-size'))
+                    data_map = {
+                        'name': data.get('name', ''),
+                        'storage_id': storage_id,
+                        'native_storage_pool_id': data.get('serial-number',
+                                                           ''),
+                        'description': '',
+                        'status': status,
+                        'storage_type': constants.StorageType.BLOCK,
+                        'total_capacity': int(total_size),
+                        'subscribed_capacity': int(blocks),
+                        'used_capacity': volume_size,
+                        'free_capacity': int(total_size - volume_size)
+                    }
+                pool_arr.append(data_map)
```

Comment:
`data_map` might be referenced before assignment.

Reply:
The next version has been modified